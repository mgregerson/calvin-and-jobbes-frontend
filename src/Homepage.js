import "./Homepage.css";

/** Homepage
 *
 * Props: user {username, firstName, lastName, email, applications: []}
 * State: None
 *
 * App -> Homepage
 */

function Homepage({ user }) {
  return (
    <div className="Homepage">
      <div className="container text-center my-auto">
        <h1 className="mb-4 fw-bold">Jobber the Hutt</h1>
        {user ? (
          <p className="lead">Welcome, {user.username}!</p>
        ) : (
          <p className="lead">Find Your Dream Job</p>
        )}
      </div>
    </div>
  );
}

export default Homepage;
