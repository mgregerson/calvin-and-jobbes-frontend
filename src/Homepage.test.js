import { render } from "@testing-library/react";
import Homepage from "./Homepage";

const TEST_USER = {
  username: "test-username",
};

describe("Homepage", function () {
  it("renders without crashing", function () {
    render(<Homepage user={TEST_USER} />);
  });
  it("matches snapshot", function () {
    const { container } = render(<Homepage user={TEST_USER} />);
    expect(container).toMatchSnapshot();
  });
  it("Displays the name of the application on the page", function () {
    const { container } = render(<Homepage user={TEST_USER} />);
    expect(container.querySelector(".Homepage-title").innerHTML).toEqual(
      "Jobber the Hutt"
    );
  });
  it("Displays a welcome message with the user's username if they are logged in", function () {
    const { container } = render(<Homepage user={TEST_USER} />);
    expect(container.querySelector(".Homepage-welcome").innerHTML).toEqual(
      "Welcome, test-username!"
    );
  });
  it("Does not display the company slogan if the user is logged in", function () {
    const { container } = render(<Homepage user={TEST_USER} />);
    expect(container.querySelector(".Homepage-slogan")).not.toBeInTheDocument();
  });
  it("Displays the company slogan if there is no logged in user provided", function () {
    const { container } = render(<Homepage />);
    expect(
      container.querySelector(".Homepage-welcome")
    ).not.toBeInTheDocument();
  });
  it("Does not display a welcome message if there is no logged in user", function () {
    const { container } = render(<Homepage />);
    expect(container.querySelector(".Homepage-slogan").innerHTML).toEqual(
      "Find Your Dream Job"
    );
  });
});
