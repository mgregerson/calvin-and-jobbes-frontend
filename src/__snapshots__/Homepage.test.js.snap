// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage matches snapshot 1`] = `
<div>
  <div
    class="Homepage"
  >
    <div
      class="container text-center my-auto"
    >
      <h1
        class="Homepage-title mb-4 fw-bold"
      >
        Jobber the Hutt
      </h1>
      <p
        class="Homepage-welcome lead"
      >
        Welcome, 
        test-username
        !
      </p>
    </div>
  </div>
</div>
`;
